### ISSUE TITLE
As of v1.5.0 karma-browserstack-launcher often times out

### ISSUE THREAD

**Issue by brandonocasey** [karma-runner/karma-browserstack-launcher#152]
Mostly on our end we run `karma-browserstack-launcher` via travis ci. When two builds, one for a pull request and one for a branch run at the same time, One of the builds will often timeout. I think the one that starts first times out as soon as the other build starts to run. On previous versions of `karma-browserstack-launcher` this did not happen.

I haven't had time to look into this too much, but it is probably a result of the switch to `browserstack-local`

**Comment by steel** [comment]
We're noticing this too. Downgrading back to 1.4.0 seems to fix the issue.

**Comment by XhmikosR** [comment]
We are hitting this on Bootstrap too. I haven't been able to pinpoint the cause yet myself.

**Comment by johnjbarton** [comment]
1.4 does not help us.  I reached out to browserstack help but they referred me to this plugin.

**Comment by XhmikosR** [comment]
The commit you referenced above does not restrain the package. It will
still use 1.5.x.

In our case the issue still happens but it's less frequent with 1.4.0
compared to 1.5.x. It does still happen unfortunately.

**Comment by andrewmclagan** [comment]
Also seeing this issue... only when on local not running in CI

**Comment by lina128** [comment]
We also have this issue.

**Comment by lina128** [comment]
In our case, we have a very large test suite. This PR that prevent browser inactive looks promising: https://github.com/karma-runner/karma-browserstack-launcher/pull/168

When is it going to be released?

**Comment by Borewit** [comment]
I never managed to get the unit test running stable via karma-browserstack. After being idle a while (not sure how long, maybe an hour) there is reasonable probability it works. It looks like after a few runs, it is more likely to fail.

**Comment by JakeChampion** [comment]
> 1.4 does not help us. I reached out to browserstack help but they referred me to this plugin.

This PR fixes the issue, are you or @dignifiedquire able to merge and release it? https://github.com/karma-runner/karma-browserstack-launcher/pull/158

**Comment by lina128** [comment]
We also experienced the same recently. Upgraded to 1.5.1, and seeing a lot of timeout. Hope this PR can fix the problem.

**Comment by XhmikosR** [comment]
@johnjbarton does https://github.com/karma-runner/karma/pull/3388 mean that this issue is fixed now with karma v5.0.0?

**Comment by johnjbarton** [comment]
no, I am not happy with the number of times that CI fails on browserstack.  But the version of karma-browserstack-launcher did not seem to have a huge effect.

I have worked quite a bit on our CI now and hope to have a clearer picture.

**Comment by XhmikosR** [comment]
Ah, wait, actually https://github.com/karma-runner/karma/pull/3388 is about a CI fix in karma itself, not this issue here.

**Comment by XhmikosR** [comment]
@johnjbarton I finally managed to get to use the BrowserStack recommended karma settings and also updated to the latest karma-browserstak-launcher version on Bootstrap: https://github.com/twbs/bootstrap/pull/30720.

It does work well, but there are still some cases the build might timeout. Probably fewer, but still.

**Comment by johnjbarton** [comment]
Yes also my experience.

**Comment by XhmikosR** [comment]
I reverted the patch and I sent BrowserStack support an email. Something is still wrong somewhere when many CI runs fire up simultaneously or too fast.

**Comment by jeremad** [comment]
someone tried it with v1.6.0?

**Comment by sculpt0r** [comment]
> someone tried it with v1.6.0?

Yes, and still often receive timeouts. I'm not sure if the issue is on the launcher side, or the Browserstack?
